bundles/ch.elexis.core.ui.e4/src/ch/elexis/core/ui/e4/dialog/MandantSelectorDialog.java

```diff
@@ -0,0 +1,163 @@
+	public Control createDialogArea(final Composite parent) {
+
+		Composite radioComposite = new Composite(parent, SWT.NONE);
+		radioComposite.setLayout(new GridLayout(2, false));
+		Button btnAlle = new Button(radioComposite, SWT.NONE);
+		btnAlle.setText("alle");
+		btnAlle.addListener(SWT.Selection, e -> {
+			uiList.getList().selectAll();
+		});
+		Button btnKeine = new Button(radioComposite, SWT.NONE);
+		btnKeine.setText("keine");
+		btnKeine.addListener(SWT.Selection, e -> {
+			uiList.getList().deselectAll();
+		});
+
+		if (multi) {
+			setTitle("Mandanten Auswahl");
+			StringBuilder msg = new StringBuilder("Bitte wählen Sie die Mandanten aus.");
+			if (noEmpty) {
+				msg.append("\nKeine Auswahl selektiert den aktiven Mandanten");
+			}
+			setMessage(msg.toString());
+			uiList = new ListViewer(parent, SWT.BORDER | SWT.MULTI);
+		} else {
+			setTitle("Mandant ändern");
+			setMessage("Bitte wählen Sie einen Mandanten");
+			uiList = new ListViewer(parent, SWT.BORDER | SWT.SINGLE);
+		}
+		mandatorsList = getMandators();
+
+		uiList.setContentProvider(ArrayContentProvider.getInstance());
+		uiList.setInput(mandatorsList);
+		uiList.getList().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		uiList.setLabelProvider(new LabelProvider() {
+			@Override
+			public String getText(Object element) {
+				return ((IMandator) element).getLabel();
+			}
+		});
+
+		setSelected(idList);
+
+		return uiList.getList();
+	}
+
+	@Override
+	protected void okPressed() {
+		Object[] selectedMandators = uiList.getStructuredSelection().toArray();
+		selMandators = new ArrayList<IMandator>();
+
+		if (selectedMandators.length > 0) {
+			for (Object element : selectedMandators) {
+				if (element instanceof IMandator) {
+					selMandators.add((IMandator) element);
+				}
+			}
+		}
+
+		if (!multi && !selMandators.isEmpty()) {
+			selMandator = selMandators.get(0);
+		}
+
+		super.okPressed();
+	}
+
+	public IMandator getSelectedMandator() {
+		return selMandator;
+	}
+	
+	public List<IMandator> getSelectedMandators() {
+		return selMandators;
+	}
+
+	private List<IMandator> getMandators() {
+		IQuery<IMandator> query = CoreModelServiceHolder.get().getQuery(IMandator.class);
+		List<IMandator> mandators = query.execute();
+
+		List<IMandator> list = mandators.stream().filter(m -> (nonActive || isActive(m)) && isMandator(m)).map(m -> m)
+				.collect(Collectors.toList());
+		list.sort(Comparator.comparing(IMandator::getLabel));
+
+		return list;
+	}
+
+	public void setSelected(String idList) {
+		if (idList.isBlank()) {
+			return;
+		}
+
+		Map<String, IMandator> mandatorMap = mandatorsList.stream().collect(Collectors.toMap(IMandator::getId, m -> m));
+
+		Arrays.stream(idList.split(",")) //$NON-NLS-1$
+				.map(String::trim).filter(mandatorMap::containsKey).forEach(id -> {
+					IMandator mandator = mandatorMap.get(id);
+					int index = mandatorsList.indexOf(mandator);
+					if (index != -1)
+						uiList.getList().select(index);
+				});
```

Comment:
Für wäre hier die Logik klarer, wenn die Liste aller Mandanten mit den ids gefiltert wird, und dann das Resultat als Selection zu setzen.

So in etwa:
`uiList.setSelection(new StructuredSelection(getMandators().stream().filter(m -> idList.contains(m.getId())).tiList())));`